src/features/swap/SwapForm.tsx

```diff
@@ -209,12 +212,16 @@ function AmountField({
 }
 
 function ReverseTokenButton() {
-  const { values, setFieldValue } = useFormikContext<SwapFormValues>()
+  const { values, setValues } = useFormikContext<SwapFormValues>()
   const { fromTokenId, toTokenId } = values
 
   const onClickReverse = () => {
-    setFieldValue('fromTokenId', toTokenId)
-    setFieldValue('toTokenId', fromTokenId)
+    setValues({
+      ...values,
+      toTokenId: fromTokenId,
+      fromTokenId: toTokenId,
+      submitType: 'reverse',
+    })
```

Comment:
I think actually all we needed to do is do an `evt.stopPropagation` here, and not the whole submitType shenanigans. @denviljclarke can you take this back?